openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -467,13 +467,8 @@ def post(self, request, *args, **kwargs):
         if response:
             return response
 
-        if 'is_account_recovery' in request.GET:
-            self.post_reset_login = True
```

Comment:
@zamanafzal according to https://docs.djangoproject.com/en/1.11/topics/auth/default/ 
```
post_reset_login: A boolean indicating if the user should be automatically authenticated after a successful password reset. Defaults to False.
```

Looks like we changing the behaviour?

Reply:
yes, you are right. But in this code, we were authenticating only when password request has been made via recovery email address. If the primary email address has been added then user is not being authenticated. This was not the right behaviour.
This code was changed in the recent past and when I look into the previous code, then there is nothing like this and we were not authenticating the user after password reset.
I am attaching the link of these previous changes herein.
[PR](https://github.com/edx/edx-platform/pull/22784)

Reply:
I agree, Zaman. I don't know why we were only authenticating after changing the password via recovery, and not when changing the password for the primary account, but your change does make the behavior consistent. In the future, we may find that we want to authenticate for both paths, at which time we will need to ensure that a single authentication backend is configured to avoid the exception we had been seeing here.